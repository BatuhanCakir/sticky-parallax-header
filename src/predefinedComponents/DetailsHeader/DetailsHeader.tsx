import React, { ReactElement } from 'react';
import {
  Text,
  View,
  Image,
  TouchableOpacity,
  StatusBar,
  Animated,
  ImageSourcePropType,
} from 'react-native';
import StickyParallaxHeader, {
  DetailsData,
  StickyParallaxHeaderProps,
} from '../../StickyParallaxHeaderComponent';
import { constants, sizes } from '../../constants';
import { Brandon } from '../../assets/data/cards';
import styles from './DetailsHeader.styles';
import { renderContent } from './defaultProps/defaultProps';
import type { IconProps, RenderBody, SharedPredefinedHeaderProps } from '../../index';

const { event, ValueXY } = Animated;

export interface DetailsHeaderProps extends IconProps, SharedPredefinedHeaderProps, RenderBody {
  headerType: 'DetailsHeader';
  hasBorderRadius?: boolean;
  iconNumber?: number;
  image?: ImageSourcePropType;
  tag?: string;
  title?: string;
}
type State = {
  headerLayout: {
    height: number;
  };
};

class DetailsHeader extends React.Component<DetailsHeaderProps, State> {
  scrollY: Animated.ValueXY;

  constructor(props: DetailsHeaderProps) {
    super(props);
    this.state = {
      headerLayout: {
        height: 0,
      },
    };

    this.scrollY = new ValueXY();
  }

  setHeaderSize: StickyParallaxHeaderProps['headerSize'] = (headerLayout) => {
    const { headerSize } = this.props;

    if (headerSize) headerSize(headerLayout);
    this.setState({ headerLayout });
  };

  scrollPosition = (value: number): number => {
    const { headerLayout } = this.state;

    return constants.scrollPosition(headerLayout.height, value);
  };

  renderHeader = (user: DetailsData): ReactElement => {
    const { backgroundColor, leftTopIconOnPress, rightTopIconOnPress, leftTopIcon, rightTopIcon } =
      this.props;

    const opacity = this.scrollY.y.interpolate({
      inputRange: [0, this.scrollPosition(60), this.scrollPosition(90)],
      outputRange: [0, 0, 1],
      extrapolate: 'clamp',
    });

    return (
      <View style={[styles.headerWrapper, { backgroundColor }]}>
        <View style={styles.headerMenu}>
          <TouchableOpacity hitSlop={sizes.hitSlop} onPress={leftTopIconOnPress}>
            {/*@ts-ignore there is default value for that*/}
            <Image style={styles.icon} resizeMode="contain" source={leftTopIcon} />
          </TouchableOpacity>
          <Animated.View style={[styles.headerTitleContainer, { opacity }]}>
            <Text style={styles.headerTitle}>{user.label}</Text>
          </Animated.View>
          <TouchableOpacity hitSlop={sizes.hitSlop} onPress={rightTopIconOnPress}>
            {/*@ts-ignore there is default value for that*/}

            <Image style={styles.icon} resizeMode="contain" source={rightTopIcon} />
          </TouchableOpacity>
        </View>
      </View>
    );
  };

  renderDetailsForeground = (user: DetailsData) => {
    const { labelColor } = user;
    const { image, title, tag, iconNumber } = this.props;
    const labelOpacity = this.scrollY.y.interpolate({
      inputRange: [0, this.scrollPosition(19), this.scrollPosition(25)],
      outputRange: [1, 0.8, 0],
      extrapolate: 'clamp',
    });
    const titleOpacity = this.scrollY.y.interpolate({
      inputRange: [0, this.scrollPosition(45), this.scrollPosition(55)],
      outputRange: [1, 0.8, 0],
      extrapolate: 'clamp',
    });
    const authorOpacity = this.scrollY.y.interpolate({
      inputRange: [0, this.scrollPosition(55), this.scrollPosition(70)],
      outputRange: [1, 0.8, 0],
      extrapolate: 'clamp',
    });

    return (
      <View style={styles.foreground}>
        <Animated.View style={[styles.foregroundTitle, { opacity: labelOpacity, labelColor }]}>
          <Text style={styles.foregroundText}>{tag}</Text>
        </Animated.View>
        <Animated.View style={[styles.messageContainer, { opacity: titleOpacity }]}>
          <Text numberOfLines={3} style={styles.message}>
            {title}
          </Text>
        </Animated.View>
        <Animated.View style={[styles.infoContainer, { opacity: authorOpacity }]}>
          <View style={styles.iconContainer}>
            <Image source={require('../../assets/icons/cards_black.png')} style={styles.icon} />
            <Text style={styles.number}>{iconNumber}</Text>
          </View>
          <View style={styles.footerContainer}>
            {/*@ts-ignore there is default value for that*/}
            <Image source={image} style={styles.authorPhoto} resizeMode="contain" />
            <Text numberOfLines={1} style={styles.authorName}>
              {user.author}
            </Text>
          </View>
        </Animated.View>
      </View>
    );
  };

  renderForeground = (user: DetailsData) => {
    const { foreground } = this.props;

    return foreground?.() ?? this.renderDetailsForeground(user);
  };

  renderBackground = () => {
    const { hasBorderRadius, backgroundColor } = this.props;
    const {
      headerLayout: { height },
    } = this.state;
    const headerBorderRadius =
      hasBorderRadius &&
      this.scrollY.y.interpolate({
        inputRange: [0, height],
        outputRange: [80, 0],
        extrapolate: 'extend',
      });

    return (
      <Animated.View
        style={[
          styles.background,
          {
            borderBottomRightRadius: headerBorderRadius,
            backgroundColor,
          },
        ]}
      />
    );
  };

  render() {
    const user = Brandon;
    const {
      backgroundColor,
      backgroundImage,
      renderBody,
      children,
      headerHeight,
      snapToEdge,
      bounces,
      parallaxHeight,
      transparentHeader,
      onMomentumScrollEnd,
      onMomentumScrollBegin,
      scrollRef,
      contentContainerStyles,
      keyboardShouldPersistTaps,
      refreshControl,
    } = this.props;

    if (renderBody) {
      console.warn('Warning: renderBody prop is deprecated. Please use children instead');
    }

    return (
      <>
        <StatusBar barStyle="light-content" backgroundColor={backgroundColor} translucent />
        <StickyParallaxHeader
          scrollEvent={event([{ nativeEvent: { contentOffset: { y: this.scrollY.y } } }], {
            useNativeDriver: false,
          })}
          background={this.renderBackground()}
          backgroundImage={backgroundImage}
          bounces={bounces}
          contentContainerStyles={contentContainerStyles}
          foreground={this.renderForeground(user)}
          header={this.renderHeader(user)}
          headerHeight={headerHeight}
          headerSize={this.setHeaderSize}
          keyboardShouldPersistTaps={keyboardShouldPersistTaps}
          onMomentumScrollBegin={onMomentumScrollBegin}
          onMomentumScrollEnd={onMomentumScrollEnd}
          parallaxHeight={parallaxHeight}
          refreshControl={refreshControl}
          scrollRef={scrollRef}
          snapToEdge={snapToEdge}
          transparentHeader={transparentHeader}>
          {renderBody ? renderBody() : children}
        </StickyParallaxHeader>
      </>
    );
  }

  static defaultProps = {
    // default is used remember to check before removing
    leftTopIcon: require('../../assets/icons/iconCloseWhite.png'),
    // default is used remember to check before removing
    rightTopIcon: require('../../assets/icons/Icon-Menu.png'),
    backgroundColor: Brandon.color,
    headerHeight: sizes.cardScreenHeaderHeight,
    backgroundImage: null,
    tag: Brandon.type,
    title: Brandon.label,
    // default is used remember to check before removing
    image: Brandon.image,
    children: renderContent(Brandon),
    bounces: true,
    snapToEdge: true,
    hasBorderRadius: true,
    iconNumber: Brandon.cardsAmount,
    parallaxHeight: sizes.cardScreenParallaxHeader,
    transparentHeader: false,
    headerSize: undefined,
    scrollRef: null,
    onMomentumScrollEnd: undefined,
    onMomentumScrollBegin: undefined,
  };
}

export default DetailsHeader;
