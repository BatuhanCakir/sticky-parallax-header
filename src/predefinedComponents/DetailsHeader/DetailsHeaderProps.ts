import type { ColorValue, ImageSourcePropType } from 'react-native';

import type {
  StickyHeaderFlatListProps,
  StickyHeaderScrollViewProps,
  StickyHeaderSectionListProps,
} from '../../primitiveComponents/StickyHeaderProps';
import type { IconProps, SharedPredefinedProps } from '../common/SharedProps';

export interface DetailsHeaderSharedProps extends IconProps, SharedPredefinedProps {
  contentIcon?: ImageSourcePropType;
  contentIconNumber?: number;
  hasBorderRadius?: boolean;
  image?: ImageSourcePropType;
  tabsContainerBackgroundColor?: ColorValue;
  tag?: string;
  title?: string;
}

export interface DetailsHeaderScrollViewProps
  extends DetailsHeaderSharedProps,
    StickyHeaderScrollViewProps {}

export interface DetailsHeaderFlatListProps<ItemT>
  extends DetailsHeaderSharedProps,
    StickyHeaderFlatListProps<ItemT> {}

export interface DetailsHeaderSectionListProps<ItemT, SectionT>
  extends DetailsHeaderSharedProps,
    StickyHeaderSectionListProps<ItemT, SectionT> {}
