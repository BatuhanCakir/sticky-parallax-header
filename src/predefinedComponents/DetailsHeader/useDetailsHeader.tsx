import React, { useCallback } from 'react';
import { Extrapolate, interpolate, useAnimatedStyle } from 'react-native-reanimated';

import type { ScrollComponent } from '../common/SharedProps';
import { HeaderWrapper } from '../common/components/HeaderWrapper';
import { usePredefinedHeader } from '../common/hooks/usePredefinedHeader';
import { scrollPosition } from '../common/utils/scrollPosition';

import type { DetailsHeaderScrollViewProps } from './DetailsHeaderProps';
import { Foreground } from './components/HeaderForeground';

export function useDetailsHeader<T extends ScrollComponent>(props: DetailsHeaderScrollViewProps) {
  const {
    contentBackgroundColor,
    onMomentumScrollEnd,
    onScroll,
    onScrollEndDrag,
    parallaxHeight,
    scrollHeight,
    scrollValue,
    scrollViewRef,
  } = usePredefinedHeader<T>(props);
  const {
    backgroundColor,
    backgroundImage,
    contentIcon,
    contentIconNumber,
    contentIconNumberStyle,
    hasBorderRadius,
    image,
    tabsContainerBackgroundColor,
    tag,
    tagStyle,
    title,
    titleStyle,
  } = props;

  const headerTitleInputRange = [
    0,
    scrollPosition(scrollHeight, 60),
    scrollPosition(scrollHeight, 90),
  ];
  const headerTitleContainerAnimatedStyle = useAnimatedStyle(() => {
    return {
      opacity: interpolate(scrollValue.value, headerTitleInputRange, [0, 0, 1], Extrapolate.CLAMP),
    };
  });

  const renderHeader = useCallback(() => {
    return (
      <HeaderWrapper
        backgroundColor={backgroundColor}
        backgroundImage={backgroundImage}
        contentBackgroundColor={contentBackgroundColor}
        hasBorderRadius={hasBorderRadius}
        parallaxHeight={parallaxHeight}
        scrollHeight={scrollHeight}
        scrollValue={scrollValue}
        tabsContainerBackgroundColor={tabsContainerBackgroundColor}>
        <Foreground
          height={parallaxHeight}
          scrollValue={scrollValue}
          contentIcon={contentIcon}
          contentIconNumber={contentIconNumber}
          contentIconNumberStyle={contentIconNumberStyle}
          image={image}
          tag={tag}
          tagStyle={tagStyle}
          title={title}
          titleStyle={titleStyle}
        />
      </HeaderWrapper>
    );
  }, [
    backgroundColor,
    backgroundImage,
    contentBackgroundColor,
    contentIcon,
    contentIconNumber,
    contentIconNumberStyle,
    hasBorderRadius,
    image,
    parallaxHeight,
    scrollHeight,
    scrollValue,
    tabsContainerBackgroundColor,
    tag,
    tagStyle,
    title,
    titleStyle,
  ]);

  return {
    headerTitleContainerAnimatedStyle,
    onMomentumScrollEnd,
    onScroll,
    onScrollEndDrag,
    renderHeader,
    scrollValue,
    scrollViewRef,
  };
}
