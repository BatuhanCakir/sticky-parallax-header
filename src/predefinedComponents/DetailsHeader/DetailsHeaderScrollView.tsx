import React, { forwardRef, useImperativeHandle } from 'react';
import type { ScrollView } from 'react-native';
import { View } from 'react-native';

import commonStyles from '../../constants/screenStyles';
import { StickyHeaderScrollView } from '../../primitiveComponents/StickyHeaderScrollView';
import type { DetailsHeaderScrollViewProps } from './DetailsHeaderProps';
import { HeaderBar } from './components/HeaderBar';
import { useDetailsHeader } from './useDetailsHeader';

export const DetailsHeaderScrollView = forwardRef<ScrollView, DetailsHeaderScrollViewProps>(
  (props, ref) => {
    const {
      backgroundColor,
      children,
      contentContainerStyle,
      decelerationRate = 'fast',
      leftTopIcon,
      leftTopIconAccessibilityLabel,
      leftTopIconOnPress,
      leftTopIconTestID,
      nestedScrollEnabled = true,
      overScrollMode = 'never',
      renderHeaderBar,
      rightTopIcon,
      rightTopIconAccessibilityLabel,
      rightTopIconOnPress,
      rightTopIconTestID,
      scrollEventThrottle = 16,
      title,
      ...rest
    } = props;
    const {
      headerTitleContainerAnimatedStyle,
      onMomentumScrollEnd,
      onScroll,
      onScrollEndDrag,
      renderHeader,
      scrollViewRef,
    } = useDetailsHeader<ScrollView>(props);

    useImperativeHandle(ref, () => scrollViewRef.current as ScrollView);

    return (
      <View style={[commonStyles.container, { backgroundColor }]}>
        {renderHeaderBar ? (
          renderHeaderBar()
        ) : (
          <HeaderBar
            backgroundColor={backgroundColor}
            headerTitleContainerAnimatedStyle={headerTitleContainerAnimatedStyle}
            leftTopIcon={leftTopIcon}
            leftTopIconAccessibilityLabel={leftTopIconAccessibilityLabel}
            leftTopIconOnPress={leftTopIconOnPress}
            leftTopIconTestID={leftTopIconTestID}
            rightTopIcon={rightTopIcon}
            rightTopIconAccessibilityLabel={rightTopIconAccessibilityLabel}
            rightTopIconOnPress={rightTopIconOnPress}
            rightTopIconTestID={rightTopIconTestID}
            title={title}
          />
        )}
        <View style={commonStyles.container}>
          <StickyHeaderScrollView
            ref={scrollViewRef}
            {...rest}
            contentContainerStyle={contentContainerStyle}
            decelerationRate={decelerationRate}
            nestedScrollEnabled={nestedScrollEnabled}
            onMomentumScrollEnd={onMomentumScrollEnd}
            onScroll={onScroll}
            onScrollEndDrag={onScrollEndDrag}
            overScrollMode={overScrollMode}
            renderHeader={renderHeader}
            scrollEventThrottle={scrollEventThrottle}>
            {children}
          </StickyHeaderScrollView>
        </View>
      </View>
    );
  }
);
