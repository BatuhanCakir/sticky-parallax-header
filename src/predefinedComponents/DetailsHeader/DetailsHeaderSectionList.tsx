import type { ForwardedRef, ReactElement, RefAttributes } from 'react';
import React, { forwardRef, useImperativeHandle } from 'react';
import type { SectionList } from 'react-native';
import { View } from 'react-native';

import commonStyles from '../../constants/screenStyles';
import { StickyHeaderSectionList } from '../../primitiveComponents/StickyHeaderSectionList';
import type { DetailsHeaderSectionListProps } from './DetailsHeaderProps';
import { HeaderBar } from './components/HeaderBar';
import { useDetailsHeader } from './useDetailsHeader';

function DetailsHeaderSectionListInner<ItemT, SectionT>(
  props: DetailsHeaderSectionListProps<ItemT, SectionT>,
  ref: ForwardedRef<SectionList<ItemT, SectionT>>
) {
  const {
    backgroundColor,
    contentContainerStyle,
    decelerationRate = 'fast',
    leftTopIcon,
    leftTopIconAccessibilityLabel,
    leftTopIconOnPress,
    leftTopIconTestID,
    nestedScrollEnabled = true,
    overScrollMode = 'never',
    renderHeaderBar,
    renderSectionFooter,
    renderSectionHeader,
    rightTopIcon,
    rightTopIconAccessibilityLabel,
    rightTopIconOnPress,
    rightTopIconTestID,
    scrollEventThrottle = 16,
    sections,
    stickySectionHeadersEnabled = true,
    title,
    ...rest
  } = props;
  const {
    headerTitleContainerAnimatedStyle,
    onScroll,
    onScrollEndDrag,
    onMomentumScrollEnd,
    renderHeader,
    scrollViewRef,
  } = useDetailsHeader<SectionList<ItemT, SectionT>>(props);

  useImperativeHandle(ref, () => scrollViewRef.current as SectionList<ItemT, SectionT>);

  return (
    <View style={[commonStyles.container, { backgroundColor }]}>
      {renderHeaderBar ? (
        renderHeaderBar()
      ) : (
        <HeaderBar
          backgroundColor={backgroundColor}
          headerTitleContainerAnimatedStyle={headerTitleContainerAnimatedStyle}
          leftTopIcon={leftTopIcon}
          leftTopIconAccessibilityLabel={leftTopIconAccessibilityLabel}
          leftTopIconOnPress={leftTopIconOnPress}
          leftTopIconTestID={leftTopIconTestID}
          rightTopIcon={rightTopIcon}
          rightTopIconAccessibilityLabel={rightTopIconAccessibilityLabel}
          rightTopIconOnPress={rightTopIconOnPress}
          rightTopIconTestID={rightTopIconTestID}
          title={title}
        />
      )}
      <View style={commonStyles.container}>
        <StickyHeaderSectionList
          ref={scrollViewRef}
          {...rest}
          contentContainerStyle={contentContainerStyle}
          decelerationRate={decelerationRate}
          nestedScrollEnabled={nestedScrollEnabled}
          onMomentumScrollEnd={onMomentumScrollEnd}
          onScroll={onScroll}
          onScrollEndDrag={onScrollEndDrag}
          overScrollMode={overScrollMode}
          renderHeader={renderHeader}
          renderSectionFooter={renderSectionFooter}
          renderSectionHeader={renderSectionHeader}
          scrollEventThrottle={scrollEventThrottle}
          sections={sections}
          stickySectionHeadersEnabled={stickySectionHeadersEnabled}
        />
      </View>
    </View>
  );
}

type DetailsHeaderSectionListType = <ItemT, SectionT>(
  props: DetailsHeaderSectionListProps<ItemT, SectionT> &
    RefAttributes<SectionList<ItemT, SectionT>>
) => ReactElement;

export const DetailsHeaderSectionList = forwardRef(
  DetailsHeaderSectionListInner
) as DetailsHeaderSectionListType;
