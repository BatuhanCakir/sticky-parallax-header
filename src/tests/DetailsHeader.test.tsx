import React from 'react';
import { cleanup } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import DetailsHeader from '../predefinedComponents/DetailsHeader/DetailsHeader';
import StickyParallaxHeaderComponent from '../StickyParallaxHeaderComponent';

describe('DetailsHeader empty', () => {
  afterEach(cleanup);

  test('Snapshot for default', () => {
    const tree = renderer.create(<DetailsHeader headerType={'DetailsHeader'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Renders title prop correctly', () => {
    const component = renderer.create(
      <DetailsHeader headerType={'DetailsHeader'} title="Test 1" />
    );
    const testInstance = component.root;

    expect(testInstance.props.title).toBe('Test 1');
  });

  test('Renders headerLayout correctly', () => {
    const component = renderer.create(<DetailsHeader headerType={'DetailsHeader'} />);
    const testInstance = component.root;
    const stickyHeader = testInstance.findByType(StickyParallaxHeaderComponent);

    expect(testInstance.instance.state.headerLayout).toMatchObject({ height: 0 });
    stickyHeader.props.headerSize({ height: 1 });
    expect(testInstance.instance.state.headerLayout).toMatchObject({ height: 1 });
  });
});
