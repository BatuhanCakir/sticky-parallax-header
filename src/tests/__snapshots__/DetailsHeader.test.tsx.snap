// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsHeader empty Snapshot for default 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "height": 85,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 24,
            "paddingTop": 40,
            "width": "100%",
          },
          Object {
            "backgroundColor": undefined,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          hitSlop={
            Object {
              "bottom": 15,
              "left": 15,
              "right": 15,
              "top": 15,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "opacity": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "lineHeight": 20,
                "paddingHorizontal": 12,
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={false}
          hitSlop={
            Object {
              "bottom": 15,
              "left": 15,
              "right": 15,
              "top": 15,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
      </View>
    </View>
  </View>
  <RCTScrollView
    bounces={true}
    bouncesZoom={true}
    contentContainerStyle={
      Object {
        "backgroundColor": undefined,
        "minHeight": 1956.02,
      }
    }
    decelerationRate="fast"
    forwardedRef={[Function]}
    nestedScrollEnabled={true}
    onScroll={[Function]}
    onScrollEndDrag={[Function]}
    overScrollMode="never"
    scrollEventThrottle={1}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "height": 707.02,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 300,
                "left": 0,
                "overflow": "scroll",
                "position": "absolute",
                "right": 0,
                "top": -300,
              },
              Object {
                "backgroundColor": undefined,
              },
              Object {
                "backgroundColor": undefined,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "justifyContent": "flex-end",
                "left": 0,
                "marginBottom": 5,
                "paddingBottom": 3,
                "position": "absolute",
                "top": 0,
                "width": 750.5,
              },
              Object {
                "height": 707.02,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": undefined,
                "borderBottomRightRadius": 80,
                "height": "100%",
                "justifyContent": "flex-end",
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": undefined,
              "height": 707.02,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "flex-end",
                "paddingHorizontal": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "rgba(255, 255, 255, 0.1)",
                  "borderRadius": 18,
                  "height": 36,
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "paddingHorizontal": 12,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "opacity": 1,
                  "paddingBottom": 24,
                  "paddingTop": 24,
                }
              }
            >
              <Text
                numberOfLines={3}
                style={
                  Object {
                    "color": "white",
                    "fontSize": 75,
                    "letterSpacing": -1,
                    "lineHeight": 90,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "opacity": 1,
                  "paddingBottom": 32,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "flexDirection": "row",
                    "padding": 8,
                    "width": 56,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "black",
                      "fontSize": 16,
                      "lineHeight": 20,
                      "paddingLeft": 4,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "marginHorizontal": 24,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 16,
                      "lineHeight": 20,
                      "paddingLeft": 12,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          alwaysBounceVertical={false}
          automaticallyAdjustContentInsets={false}
          contentContainerStyle={
            Object {
              "minHeight": 541.98,
            }
          }
          contentOffset={
            Object {
              "x": 0,
            }
          }
          directionalLockEnabled={true}
          forwardedRef={[Function]}
          horizontal={true}
          keyboardDismissMode="on-drag"
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          pagingEnabled={true}
          scrollEnabled={true}
          scrollEventThrottle={16}
          scrollsToTop={false}
          showsHorizontalScrollIndicator={false}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
